src/CSharp/CodeCracker/Style/ForInArrayAnalyzer.cs

```diff
@@ -65,7 +65,18 @@ private static void Analyzer(SyntaxNodeAnalysisContext context)
                     return !identifierSymbol.Equals(arrayId);
                 });
             if (otherUsesOfIndexToken != 0) return;
-
+            var baseTypeName = (SymbolType(arrayId) as IArrayTypeSymbol)?.ElementType?.BaseType?.Name ?? "";
```

Comment:
Yes, but in this case, all types for this analyzer are Enumerable (IsEnumerable function), even so, are an array.